joblib/test/test_func_inspect.py

```diff
@@ -6,54 +6,24 @@
 # Copyright (c) 2009 Gael Varoquaux
 # License: BSD Style, 3 clauses.
 
-import os
-import shutil
-import tempfile
 import functools
 
 from joblib.func_inspect import filter_args, get_func_name, get_func_code
 from joblib.func_inspect import _clean_win_chars, format_signature
 from joblib.memory import Memory
 from joblib.test.common import with_numpy
-from joblib.testing import assert_equal, assert_raises_regex, assert_raises
+from joblib.testing import (fixture, parametrize, pytest_assert_raises)
 from joblib._compat import PY3_OR_LATER
 
 
 ###############################################################################
-# Module-level functions, for tests
+# Module-level functions and fixture, for tests
 def f(x, y=0):
     pass
 
 
-def g(x, y=1):
-    """ A module-level function for testing purposes.
-    """
-    return x ** 2 + y
-
-
-def f2(x):
-    pass
-
-
-# Create a Memory object to test decorated functions.
-# We should be careful not to call the decorated functions, so that
-# cache directories are not created in the temp dir.
-temp_folder = tempfile.mkdtemp(prefix="joblib_test_func_inspect_")
-mem = Memory(cachedir=temp_folder)
-
-
-def teardown_module():
-    if os.path.exists(temp_folder):
-        try:
-            shutil.rmtree(temp_folder)
-        except Exception as e:
-            print("Failed to delete temporary folder %s: %r" %
-                  (temp_folder, e))
-
-
-@mem.cache
 def g(x):
-    return x
+    pass
```

Comment:
Nope, this one is correct. Earlier it was:

```python
def f2(x):
    pass

@mem.cache
def g(x):
    return x
```

Just the names `g` and `f2` were interchanged because `f2` was the odd one out (decorated).